docs/source/raft_configuration.md

```diff
@@ -330,5 +330,26 @@ channels (this does not mean they share data --- they do not!), if a Raft node i
 part of the consenter set in many channels, you might want to lengthen the amount
 of time it takes to trigger an election to avoid inadvertent leader elections.
 
+2. As aforementioned, write-ahead logs (wal) and snapshots should be persisted in
+order to attain safety proerty of Raft protocol. However, it may happen if a node
+loses its data by accident, and the node would panic with following message:
+```
+tocommit(X) is out of range [lastIndex(Y)]. Was the raft log corrupted, truncated, or lost?
+```
+Human intervention is required to resurrect problematic node in this case:
+- Option 1: remove node and readd it to consenter set via config update, following
+instructions in reconfiguration section in this doc. This is the most appropriate
+option and does not violate safety property of Raft protocol.
+- Option 2: wipe out wal and snapshot directories of problematic channel, restart
+current leader node, then restart corrupted orderer. This essentially treats corrupted
+orderer as a fresh node without any data.
+- Option 3: to avoid downtime caused by restarting leader node, you could also wipe out
+wal and snapshot directories of problematic channel, and copy a snapshot from an alive node
+with greater index than X in above panic message. The index can be inferred from snapshot
+file names, for example `0000000000000002-0000000000000006.snap` stores a snapshot with
+index 6. Then start node and wait for it to catch up.
+
+Note: option 2&3 do violate safety property of Raft protocol and should be used with caution.
```

Comment:
and this makes me wonder why would be enlist 2 and 3 if both violates safety? is there a practical use case where users will choose 2 or 3 over the first one?

Reply:
the first one is fairly heavy as it affects all the channels that this node participates in. Added some text to document this reason.

Reply:
but @guoger don't you agree that option 3 could potentially cause more damage?

Reply:
@C0rWin I think the potential damage they have is the same - chain being forked in extreme cases. However the advantage that option 3 has over option 2 is reduced downtime, especially for 3 node clusters. This is the main reason i think it's worth to document option 3. wdyt?